src/network_traffic_filter.c

```diff
@@ -364,17 +402,23 @@ n2n_verdict filter_packet_from_tap(network_traffic_filter_impl_t *filter, n2n_ed
   return N2N_ACCEPT;
 }
 
+/* for [-Wmissing-declarations] */
+network_traffic_filter_t *create_network_traffic_filter();
+
 network_traffic_filter_t *create_network_traffic_filter() {
   network_traffic_filter_impl_t *filter = malloc(sizeof(network_traffic_filter_impl_t));
 
   memset(filter, 0, sizeof(network_traffic_filter_impl_t));
-  filter->filter_packet_from_peer = filter_packet_from_peer;
-  filter->filter_packet_from_tap = filter_packet_from_tap;
-  return filter;
+  filter->filter_packet_from_peer = (filter_packet_from_peer_func_t)filter_packet_from_peer;
```

Comment:
Why do you need a cast here? Can you please define filter_packet_from_peer of type filter_packet_from_peer_func_t instead of casting?
Same applies to filter_packet_from_tap

Reply:
I want to expose `network_traffic_filter_t` which contain interface  to outside, but hide implementation details inside `network_traffic_filter_impl_t`.  Codes outside `network_traffic_filter.c/h` no need to care about fields implementation details of `network_traffic_filter_impl_t`. for example fields  `filter_rule_t *rules`,`filter_rule_pair_cache_t *connections_rule_cache` inside `network_traffic_filter_impl_t` shouldn't accessible outside `network_traffic_filter.c/h`.

If this does not conform to the coding specification of the project, I will change it later.